templates/skeleton/app/root.tsx

```diff
@@ -78,10 +88,10 @@ export async function loader({context}: LoaderFunctionArgs) {
 
   return defer(
     {
-      cart: cartPromise,
-      footer: footerPromise,
-      header: await headerPromise,
-      isLoggedIn: isLoggedInPromise,
+      cart: cartPromise as Promise<CartApiQueryFragment | null>,
+      footer: footerPromise as Promise<FooterQuery>,
+      header: (await headerPromise) as HeaderQuery,
+      isLoggedIn: isLoggedInPromise as Promise<boolean>,
```

Comment:
Ideally we wouldn't need to cast types here... is this required to fix something?